substrate/bits

```diff
@@ -51,12 +51,16 @@ namespace substrate
 		);
 	}
 
-	/* TODO: Determin if we can drop this and abstract rotl/rotr to shift by the fixed nybble amount */
+	/* TODO: Determine if we can drop this and abstract rotl/rotr to shift by the fixed nibble amount */
 	template<typename T> SUBSTRATE_NO_DISCARD(typename std::enable_if<std::is_integral<T>::value ,T>::type
-		constexpr inline shift_nibble(const T v, const std::size_t shift) noexcept)
+		constexpr inline shift_nibble(const T value, const std::size_t shift) noexcept)
 	{
-		return (((v | 0x00U) << ((shift & ((sizeof(T) << 1) + ~0x00U)) << 2)) |
-			((v | 0x00U) >> ((sizeof(T) << 3) + 0x01U + ~((shift & ((sizeof(T) << 1) + ~0x00U)) << 2))));
+		constexpr size_t width{sizeof(T) << 3};
+		const size_t left{(shift & ((sizeof(T) << 1) + ~0x00U)) << 2};
+		const size_t right{(sizeof(T) << 3) + 0x01U + ~((shift & ((sizeof(T) << 1) + ~0x00U)) << 2)};
```

Comment:
Please suffix these shift mounts with U to ensure the operations are kept unsigned